internal/tekton-bundle/tekton-bundle.go

```diff
@@ -38,74 +36,34 @@ type Bundle struct {
 	ConfigMaps      []v1.ConfigMap
 }
 
-func ReadTasksBundle(isOpenshift bool) (*Bundle, error) {
-	var files [][]byte
-	path := getTasksBundlePath(isOpenshift)
-	data, err := os.ReadFile(path)
-	if err != nil {
-		return nil, err
-	}
-	files = append(files, data)
-
-	tektonObjs, err := decodeObjectsFromFiles(files)
-	if err != nil {
-		return nil, err
-	}
-
-	return tektonObjs, nil
+func GetTektonPipelineBundlePaths() []string {
+	var paths []string
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "pipelines-rbac.yaml"))
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "windows-bios-installer-configmaps.yaml"))
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "windows-bios-installer-pipeline.yaml"))
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "windows-customize-configmaps.yaml"))
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "windows-customize-pipeline.yaml"))
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "windows-efi-installer-configmaps.yaml"))
+	paths = append(paths, filepath.Join(tektonPipelinesBundleDir, "windows-efi-installer-pipeline.yaml"))
+	return paths
 }
 
-func ReadPipelineBundle() (*Bundle, error) {
-	path := getPipelineBundlePath()
-	files, err := readFolder(path)
-	if err != nil {
-		return nil, err
-	}
-
-	tektonObjs, err := decodeObjectsFromFiles(files)
-	if err != nil {
-		return nil, err
-	}
-
-	return tektonObjs, nil
-}
-
-func getPipelineBundlePath() string {
-	return tektonPipelinesBundleDir
-}
-
-func getTasksBundlePath(isOpenshift bool) string {
-	if isOpenshift {
+func GetTektonTasksBundlePath(isOpenShift bool) string {
+	if isOpenShift {
 		return filepath.Join(tektonTasksOKDBundleDir, "kubevirt-tekton-tasks-okd.yaml")
 	}
 	return filepath.Join(tektonTasksKubernetesBundleDir, "kubevirt-tekton-tasks-kubernetes.yaml")
 }
 
-func readFolder(folderPath string) ([][]byte, error) {
-	files, err := os.ReadDir(folderPath)
-	if err != nil {
-		return nil, err
-	}
-	filesBytes := make([][]byte, 0, len(files))
-	for _, file := range files {
-		if file.IsDir() {
-			continue
-		}
-
-		f, err := os.ReadFile(filepath.Join(folderPath, file.Name()))
+func ReadBundle(paths []string) (*Bundle, error) {
+	bundle := &Bundle{}
+	for _, path := range paths {
+		file, err := os.Open(path)
 		if err != nil {
 			return nil, err
 		}
-		filesBytes = append(filesBytes, f)
-	}
-
-	return filesBytes, nil
-}
-
-func decodeObjectsFromFiles(files [][]byte) (*Bundle, error) {
-	bundle := &Bundle{}
-	for _, file := range files {
-		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
+		defer file.Close()
```

Comment:
An unintuitive behavior of golang is that `defer` statements are executed when returning from a function, and not when a scope ends with `}`, like for example a destructor in C++ would be executed. When you call `defer` in a loop, all the defers will be remembered and only called on return. In this case, all the files will be closed only on return.

Reply:
Ack, thanks for the explanation. Good to know.

Reply:
Done. I'll leave that discussion open so others can learn as well.